gateway/ln-gateway/src/rpc/rpc_server.rs

```diff
@@ -157,6 +168,18 @@ fn v1_routes(gateway: Gateway) -> Router {
         .layer(CorsLayer::permissive())
 }
 
+/// Creates a password hash by appending a 4 byte salt to the plaintext
+/// password.
+pub fn hash_password(plaintext_password: String, salt: [u8; 16]) -> sha256::Hash {
+    let mut bytes = Vec::<u8>::new();
+    plaintext_password
+        .consensus_encode(&mut bytes)
+        .expect("Password is encodable");
+    salt.consensus_encode(&mut bytes)
+        .expect("Salt is encodable");
+    sha256::Hash::hash(&bytes)
+}
```

Comment:
Probably fine since the salt is drawn randomly, but normally one would use an HMAC imo or even specialized password hashing algorithms.